packages/builders/src/interactions/commands/chatInput/mixins/SharedChatInputCommandOptions.ts

```diff
@@ -0,0 +1,144 @@
+import { ChatInputCommandAttachmentOption } from '../options/attachment.js';
+import { ChatInputCommandBooleanOption } from '../options/boolean.js';
+import { ChatInputCommandChannelOption } from '../options/channel.js';
+import { ChatInputCommandIntegerOption } from '../options/integer.js';
+import { ChatInputCommandMentionableOption } from '../options/mentionable.js';
+import { ChatInputCommandNumberOption } from '../options/number.js';
+import { ChatInputCommandRoleOption } from '../options/role.js';
+import { ChatInputCommandStringOption } from '../options/string.js';
+import { ChatInputCommandUserOption } from '../options/user.js';
+import type { ApplicationCommandOptionBase } from './ApplicationCommandOptionBase.js';
+
+export interface SharedChatInputCommandOptions {
+	options?: ApplicationCommandOptionBase[];
+}
+/**
+ * This mixin holds symbols that can be shared in chat input command options.
+ *
+ * @typeParam TypeAfterAddingOptions - The type this class should return after adding an option.
+ */
+export class SharedChatInputCommandOptions {
+	protected declare readonly data: SharedChatInputCommandOptions;
+
+	/**
+	 * Adds a boolean option.
+	 *
+	 * @param input - A function that returns an option builder or an already built builder
+	 */
+	public addBooleanOption(
```

Comment:
Should all of these support adding multiple options at once?

Reply:
This one I'm also leaning towards a hard-no...